src/pydata_sphinx_theme/assets/styles/extensions/_sphinx_design.scss

```diff
@@ -71,3 +78,90 @@
     }
   }
 }
+/*******************************************************************************
+ * color overrides. Here we recreate the sphinx-design rules for colors
+ * (including button and badge colors), and assign them our own --pst-color-*
+ * variables instead of --sd-color-*.
+ *
+ * First, define the extra keys needed to cover the full range of semantic
+ * color names used in sphinx-design, then merge them with the names we
+ * already define for our own needs.
+ */
+$extra-semantic-colors: (
+  "dark": map-get($pst-semantic-colors, "text-base"),
+  "muted": map-get($pst-semantic-colors, "text-muted"),
+  "light": rgb(201, 201, 201),
+  "black": rgb(0, 0, 0),
+  "white": rgb(255, 255, 255),
+);
+
+$all-colors: map-merge($pst-semantic-colors, $extra-semantic-colors);
+
+/* Next, make a list of the semantic color names from sphinx-design (we only
+ * need to create override rules for classes that sphinx-design has defined).
+ * Here we also incorporate a modified version of sphinx-design's text-color
+ * function, to make sure that text is legible on colored buttons and badges.
+ */
+$sd-semantic-color-names: (
+  "primary",
+  "secondary",
+  "success",
+  "info",
+  "warning",
+  "danger",
+  "light",
+  "muted",
+  "dark",
+  "black",
+  "white"
+);
+
+@function text-color($value) {
+  @return if(lightness($value) > 70, #000, #fff);
+}
+
+// Now we define the variables and create the override rules.
```

Comment:
I did try that first, and it didn't work. It's a scope issue I think. The SD rules get... evaluated?... in a context where our overrides haven't happened yet, so we have to rewrite the rule and rely on the browser to pick our rule instead of theirs.